lib/rubocop/cop/rspec/nested_groups.rb

```diff
@@ -97,30 +97,34 @@ class NestedGroups < Cop
           "Configuration key `#{DEPRECATED_MAX_KEY}` for #{cop_name} is " \
           'deprecated in favor of `Max`. Please use that instead.'
 
-        def_node_search :find_contexts, ExampleGroups::ALL.block_pattern
+        def_node_search :find_example_groups, ExampleGroups::ALL.block_pattern
+        def_node_matcher :example_group?, ExampleGroups::ALL.block_pattern
 
         def on_top_level_describe(node, _args)
-          find_nested_contexts(node.parent) do |context, nesting|
+          find_nested_example_groups(node.parent) do |example_group, nesting|
             self.max = nesting
```

Comment:
I wonder if this works correctly, e.g. if there are two example groups on the first level, and the first one has a larger number of nested groups, would the max be set to that number, or to the nesting level of the last offending group?

Reply:
It should work correctly as far as we don't track nesting level and calculate it for every time:

```ruby
        def nesting_count(node)
          count = node.each_ancestor(:block).count { |n| example_group?(n) }
          count + 1
        end
```

Reply:
Maybe I don't understand you well. What in your opinion may work incorrectly?

Reply:
Hmm, not sure how this `self.max` value is used but looks like setting a value smaller that previous one is handled correctly:

```ruby
      def max=(value)
        cfg = config_to_allow_offenses
        cfg[:exclude_limit] ||= {}
        current_max = cfg[:exclude_limit][max_parameter_name]
        value = [current_max, value].max if current_max.                # <====
        cfg[:exclude_limit][max_parameter_name] = value
      end
```

https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/mixin/configurable_max.rb#L10-L16

Reply:
Yes, it's used when you run `--auto-gen-config`
Thank you for checking it though. I was going to verify myself, as it's not in the scope of the PR but I just noticed while reviewing the code